src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -1,473 +1,113 @@
-open Core_kernel
 open Async_kernel
-open Protocols.Coda_pow
-open Protocols.Coda_transition_frontier
 open Coda_base
-open Signature_lib
+open Core_kernel
+open Protocols.Coda_pow
 
 module type Inputs_intf = sig
-  module Staged_ledger_aux_hash : Staged_ledger_aux_hash_intf
+  include Transition_frontier0.Inputs_intf
+end
+
+module Make (Inputs : Inputs_intf) :
+  Transition_frontier_intf
+  with type state_hash = State_hash.t
+   and type external_transition_verified =
+              Inputs.External_transition.Verified.t
+   and type ledger_database = Ledger.Db.t
+   and type staged_ledger_diff = Inputs.Staged_ledger_diff.t
+   and type staged_ledger = Inputs.Staged_ledger.t
+   and type masked_ledger = Ledger.Mask.Attached.t
+   and type transaction_snark_scan_state = Inputs.Staged_ledger.Scan_state.t =
+struct
+  type state_hash = State_hash.t
+
+  type external_transition_verified = Inputs.External_transition.Verified.t
+
+  type ledger_database = Ledger.Db.t
+
+  type staged_ledger_diff = Inputs.Staged_ledger_diff.t
 
-  module Ledger_proof_statement :
-    Ledger_proof_statement_intf with type ledger_hash := Frozen_ledger_hash.t
+  type staged_ledger = Inputs.Staged_ledger.t
 
-  module Ledger_proof : sig
-    include
-      Ledger_proof_intf
-      with type statement := Ledger_proof_statement.t
-       and type ledger_hash := Frozen_ledger_hash.t
-       and type proof := Proof.t
-       and type sok_digest := Sok_message.Digest.t
+  type masked_ledger = Ledger.Mask.Attached.t
```

Comment:
prefer `:=` to `=` on these types in the signature so you don't need to duplicate the definitions (and it makes merlin easier to use) -- see the old version of this file before your diff.

Reply:
I was originally having issues getting that to be happy, I'll show you tomorrow if I can't get it working.